### ISSUE TITLE
/home/mjb/Pluralith/bin/Pluralith.AppImage: no such file

### ISSUE THREAD

**Issue by mjbright** [Pluralith/pluralith-cli#87]
Running on Ubuntu 20.04 LTS on linux_amd64, after performing ```terraform init``` (v1.2.1) and ```pluralith init```

```
pluralith plan
⠿ Running Plan ⇢ Apply in the Pluralith UI

✘ Failed to launch Pluralith
launching Pluralith failed -> RunTerraform: could not run launch command -> LaunchPluralith: runOsCommand: fork/exec /home/mjb/Pluralith/bin/Pluralith.AppImage: no such file or directory
```

Files under ~/Pluralith:
```
find ~/Pluralith/
/home/mjb/Pluralith/
/home/mjb/Pluralith/credentials
/home/mjb/Pluralith/bin
/home/mjb/Pluralith/bin/pluralith-cli-graphing
```

**Comment by DanThePutzer** [comment]
Hi @mjbright, did you download the CLI directly from the latest release on this repo?

That currently doesn't get you the desktop app, which is required. The current install flow is supposed to be the following:
1. Download the Linux installer from our website [here](https://www.pluralith.com/thanks)
2. Run the installer (chmod +x it and then run like a regular bash script)
3. Afterwards the desktop app should open automatically and going through the setup wizard in the UI would install the CLI

I realize this is not optimal, especially for Linux (we've had some issues with our Linux installer) and we'll have to improve this.

Probably a better flow for Linux users would be to have them install the CLI and add it to PATH, then have the CLI install the desktop app.

Anyways, if you want to give our Linux installer a go please download it [here](https://www.pluralith.com/thanks)

Let me know if you run into any issues that way!

**Comment by mjbright** [comment]
Yes, I just installed from the releases page ... sorry, force of habit.

So how can I download the Linux installer from the command-line ?

I need to download in an ssh session on a headless Linux machine - but I'm on macOs, so clicking the link downloads the macOS installer.

**Comment by DanThePutzer** [comment]
You're actually making a good point here. We do need to change that install process. The UI should be installable via the CLI. That way people can just grab the latest CLI release and go from there.

Regarding getting Pluralith on a headless machine: Actually simply downloading the CLI from the latest release is the best way to go here. Instead of `pluralith plan` (which works with the desktop app) you can run `pluralith graph`. `pluralith graph` is specifically for headless use. Here's some more info on it: https://docs.pluralith.com/docs/cli/graph

One thing to note: You'll need an API key to generate diagrams headlessly. You already have an account if I remember correctly from what you told @PhiWeber, so you can simply grab the API key in the dashboard or the desktop app by clicking on the little profile icon in the top right corner. (In the dashboard you'll also have to click on *User Settings*) Then you can run `pluralith login` and pass it the API key. After that `pluralith graph` should work!

Let me know if this works on your end!

**Comment by jghal** [comment]
I can't seem to find how to download the linux installer from a windows desktop (to use in WSL).

**Comment by DanThePutzer** [comment]
Hi @jghal 

Actually if you download the Windows version it should (in a perfect world) detect WSL and automatically install the Linux binary into it.

Did you install on Windows already?

**Comment by jghal** [comment]
Hi Dan,

Rather than installing the CLI, the Windows installer seems to launch the GUI and then the GUI tries to download and install the CLI but it's just hanging for me (my first guess would be proxy server). I'm starting over with the 0.2.0 release and just going with the CLI binary from the releases page directly like you mentioned in another comment for headless use. That seems to be working better except it can't seem to run the installed graph module binary (which is admittedly getting into "open a new issue" territory)

```bash
generating diagram failed -> GenerateGraph: running CLI command failed -> GenerateDiagram: fork/exec /root/Pluralith/bin/pluralith-cli-graphing: no such file or directory
$ ls -l /root/Pluralith/bin/
total 50920
-rwx------    1 root     root      52138063 Feb  7 18:53 pluralith-cli-graphing
$ /root/Pluralith/bin/pluralith-cli-graphing --help
bash: /root/Pluralith/bin/pluralith-cli-graphing: No such file or directory
$ node --version
v16.11.1
```

And I've noticed that every `pluralith` command I run goes through downloading and installing the graphing binary.

**Comment by DanThePutzer** [comment]
Hi @jghal 

Let me see if I got this correctly: Running `pluralith` downloads the graph module binary, but when you try to run `pluralith graph` it still says "no such file or directory" for the binary?

**Comment by jghal** [comment]
Correct. The issue appears to be bad error messages, as using a container that has a newer version of `node` seems to resolve the issue. So it wasn't the graph binary itself that was being reported as not found. It was something related to `node` that the graph binary was attempting to execute, and that failure to execute was causing the `pluralith` binary to re-download the graph binary every time.

**Comment by sean-akatsuki** [comment]
same problem.  is this issue solved?

**Comment by ibaraki-douji** [comment]
still happpening
